hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -22,11 +24,30 @@ def create_phase_from_complex(incpx, outfloat, width):
     execute(f"cpx_to_real {incpx} {outfloat} {width} 4", uselogging=True)
 
 
-def unwrapping_geocoding(reference, secondary, step="man", rlooks=10, alooks=2, trimode=0,
+def create_water_mask(cc_mask_file, mwidth, lt, demw, demn, dempar, safe_dir):
+    """createwater_mask based on the cc_mask_file
+
+    """
+    tmp_mask = 'tmp_mask'
+
+    os.system('cp {} {}.bmp'.format(cc_mask_file, tmp_mask))
+    # 2--bmp
+    geocode_back("{}.bmp".format(tmp_mask), "{}_geo.bmp".format(tmp_mask), mwidth, lt, demw, demn, 2)
+    # 0--bmp
+    data2geotiff("{}_geo.bmp".format(tmp_mask), "{}_geo.tif".format(tmp_mask), dempar, 0)
+    # create final_water_mask.tif file
+    mask = get_water_mask("{}_geo.tif".format(tmp_mask), safe_dir, mask_value=1)
+
+    return mask
+
+
+def unwrapping_geocoding(reference_file, secondary_file, step="man", rlooks=10, alooks=2, trimode=0,
                          npatr=1, npata=1, alpha=0.6):
     dem = "./DEM/demseg"
     dempar = "./DEM/demseg.par"
     lt = "./DEM/MAP2RDC"
+    reference = reference_file[17:25]
+    secondary = secondary_file[17:25]
```

Comment:
When you're calling `unwrapping_geocoding` here:
https://github.com/ASFHyP3/hyp3-gamma/pull/283/files#diff-017857298d747c2d70d925e08073772c23e4b62be98e23476baa0de0a7d68729R373

There are already parsed as `reference_date_short` and `secondary_date_short` here:
https://github.com/ASFHyP3/hyp3-gamma/pull/283/files#diff-017857298d747c2d70d925e08073772c23e4b62be98e23476baa0de0a7d68729R295-R298

and get defined again to `reference` and `secondary`:
https://github.com/ASFHyP3/hyp3-gamma/pull/283/files#diff-017857298d747c2d70d925e08073772c23e4b62be98e23476baa0de0a7d68729R332-R333

Why not simply pass `reference_date_short` and `secondary_date_short` to this function instead of the file names?

Reply:
In side unwrapping_geocoding, It calls create_water_mask, and this function needs reference_file, This is why I input the reference_file and secondary_file in the unwrapping_geocoding.